test/e2e/performance_basic_test.go

```diff
@@ -0,0 +1,204 @@
+//go:build performance_tests
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/utils/kubernetes/generators"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/test"
+	"github.com/kong/kubernetes-ingress-controller/v3/test/internal/helpers"
+)
+
+var (
+	defaultResNum = 10000
+	resNumStr     = os.Getenv("PERF_RES_NUMBER")
+)
+
+// -----------------------------------------------------------------------------
+// E2E Performance tests
+// -----------------------------------------------------------------------------
+
+// TestBasicHTTPRoute will create a basic HTTP route and test its functionality
+// against a Kong proxy. This test will be used to measure the performance of
+// the KIC with OpenTelemetry.
+func TestBasicPerf(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, env := setupE2ETest(t)
+
+	if resNumStr != "" {
+		if num, err := strconv.Atoi(resNumStr); err == nil {
+			defaultResNum = num
+		}
+	}
+
+	t.Log("deploying kong components")
+	ManifestDeploy{Path: dblessPath}.Run(ctx, t, env)
+
+	t.Log("deploying a minimal HTTP container deployment to test Ingress routes")
+	container := generators.NewContainer("httpbin", test.HTTPBinImage, test.HTTPBinPort)
+	deployment := generators.NewDeploymentForContainer(container)
+	deployment, err := env.Cluster().Client().AppsV1().Deployments("default").Create(ctx, deployment, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeLoadBalancer)
+	_, err = env.Cluster().Client().CoreV1().Services("default").Create(ctx, service, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	kubeconfig := getTemporaryKubeconfig(t, env)
+
+	ingressTpl := `
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: test-ingress-%d
+spec:
+  ingressClassName: kong
+  rules:
+  - host: example-%d.com
+    http:
+      paths:
+      - backend:
+          service:
+            name: httpbin
+            port:
+              number: 80
+        path: /get
+        pathType: Exact
+
+`
+
+	ingressYaml := ""
+	for i := 0; i < defaultResNum; i++ {
+		ingressYaml += fmt.Sprintf(ingressTpl, i, i)
+	}
+
+	startTime := time.Now()
+	err = applyResourceWithKubectl(ctx, t, kubeconfig, ingressYaml)
```

Comment:
Instead of using a stringified template could we use a typed client? Basically [`Clientset`](https://pkg.go.dev/k8s.io/client-go/kubernetes#Clientset)'s https://pkg.go.dev/k8s.io/client-go@v0.29.1/kubernetes/typed/networking/v1#IngressesGetter.Ingresses. Like here: https://github.com/Kong/kubernetes-ingress-controller/blob/9bbee3b9908b5945bbe77d8112b6ff29dd810174/test/integration/isolated/ingress_test.go#L348.

Would that also work for us?

Reply:
This is currently done to demonstrate its feasibility. I'm modifying it and adding more other resources, such as plugins and consumers. Our goal is to verify the time required and performance of KIC when a large number of configurations are applied to Kubernetes.

So typed client is not very convenient